src/Parser.php

```diff
@@ -363,18 +358,37 @@ protected function isIpv6Host(string $ipv6): bool
      */
     protected function isRegisteredName(string $host): bool
     {
-        if ('.' === substr($host, -1, 1)) {
-            $host = substr($host, 0, -1);
+        // Note that unreserved is purposely missing . and -
+        // . is used to separate labels
+        // - cannot start or end a label
```

Comment:
There's no reason for this. The RFC allows any sequence, just DNS has that restriction, see https://tools.ietf.org/html/rfc3986#section-3.2.2

>     reg-name    = *( unreserved / pct-encoded / sub-delims )